subprojects/execution/src/main/java/org/gradle/internal/execution/history/AfterPreviousExecutionState.java

```diff
@@ -38,6 +38,5 @@
     @Override
     ImmutableSortedMap<String, FileCollectionFingerprint> getInputFileProperties();
 
-    @Override
     ImmutableSortedMap<String, FileCollectionFingerprint> getOutputFileProperties();
```

Comment:
I re-added the `@Override`. Do you mean it looks weird that we override the base methods here? I wonder we added the overrides in the first place.

Reply:
I think I know why we had the overrides here, and they should not be necessary anymore. In the past we had a hierarchy for fingerprints where `CurrentFCF` and `PreviousFCF` has a shared supertype of `FCF`, so we overrode the return types here with `PreviousFCF`. Now we only have `CurrentFCF` extend `FCF`, thus the overrides are not necessary anymore.

Reply:
If we don't add the overrides, things become a little bit weird due to the invariance of Java generics... It would probably work, though it adds a lot of `<? extends ...>` noise.

Note that we still need the `BeforeExecutionState.getInputFileProperties()` to return a `ImmutableSortedMap<String, CurrentFileCollectionFingerprint>`.

Reply:
Ah, right, we have `? extends` in the superclass.